core/app/c/[communitySlug]/types/TypeEditor.tsx

```diff
@@ -24,12 +36,23 @@ type Props = {
 const schema = z.object({
 	name: z.string(),
 	description: z.string().optional(),
-	pubFields: z.array(z.object({ fieldId: z.string(), name: z.string(), slug: z.string() })),
+	pubFields: z
+		.array(
+			z.object({
+				fieldId: z.string(),
+				name: z.string(),
+				slug: z.string(),
+				schemaName: z.nativeEnum(CoreSchemaType).nullable(),
+			})
+		)
+		.min(1, { message: "Add at least one field" }),
+	titleField: z.string().min(1, { message: "Designate a title field" }),
```

Comment:
I asked about this in Slack. I think this is a good constraint, but I'm not sure everyone agrees!

Reply:
as discussed in slack, I'll remove this constraint! also removing the "disabled" prop on `RemoveFieldButton` when the field was the title